### PR TITLE
Correct retrieving deserializer / serializer from the bean context

### PR DESCRIPTION
**Pull request by dstepanov**

### CODE REVIEW

File: serde-support/src/main/java/io/micronaut/serde/support/LegacyBeansFactory.java [hunk 1]
```diff
@@ -69,6 +72,32 @@ <S extends Serde<T>, T> S provideSerde(InjectionPoint<Serde<T>> serdeInjectionPo
...
+
+    @Any
+    @Prototype
+    @BootstrapContextCompatible
+    <S extends Serializer<T>, T> S provideSerializer(InjectionPoint<Serializer<T>> injectionPoint,
+                                                     BeanProvider<DefaultSerdeRegistry> serdeRegistry) throws SerdeException {
+        if (injectionPoint instanceof ArgumentInjectionPoint<?, ?> argumentInjectionPoint) {
+            Argument typeParameter = argumentInjectionPoint.getArgument().getTypeParameters()[0];
+            DefaultSerdeRegistry defaultSerdeRegistry = serdeRegistry.get();
+            return (S) defaultSerdeRegistry.findSerializer(typeParameter);
```
Comment (graemerocher) [inline serde-support/src/main/java/io/micronaut/serde/support/LegacyBeansFactory.java:96]: does it need a call to `createSpecific`?

File: serde-support/src/main/java/io/micronaut/serde/support/LegacyBeansFactory.java [hunk 2]
```diff
@@ -69,6 +72,32 @@ <S extends Serde<T>, T> S provideSerde(InjectionPoint<Serde<T>> serdeInjectionPo
...
 
+    @Any
+    @Prototype
+    @BootstrapContextCompatible
+    <S extends Deserializer<T>, T> S provideDeserializer(InjectionPoint<Deserializer<T>> injectionPoint,
+                                                         BeanProvider<DefaultSerdeRegistry> serdeRegistry) throws SerdeException {
+        if (injectionPoint instanceof ArgumentInjectionPoint<?, ?> argumentInjectionPoint) {
+            Argument typeParameter = argumentInjectionPoint.getArgument().getTypeParameters()[0];
+            DefaultSerdeRegistry defaultSerdeRegistry = serdeRegistry.get();
+            return (S) defaultSerdeRegistry.findDeserializer(typeParameter);
```
Comment (graemerocher) [inline serde-support/src/main/java/io/micronaut/serde/support/LegacyBeansFactory.java:83]: does it need a call to createSpecific?

File: serde-support/src/test/groovy/io/micronaut/serde/support/serdes/BeanContextSerdeSpec.groovy [hunk 3]
```diff
@@ -71,5 +73,21 @@ class BeanContextSerdeSpec extends Specification {
...
+        then:
+        deserializer
+    }
+
+    void "test retrieving Map serializer"() {
+        when:
+        def deserializer = beanContext.getBean(Argument.of(Serializer, Argument.mapOf(String, Object)))
+
+        then:
+        deserializer
```
Comment (graemerocher) [inline serde-support/src/test/groovy/io/micronaut/serde/support/serdes/BeanContextSerdeSpec.groovy]: add tests that actually use the serialiser/deserializer to serialize

**Review by graemerocher — CHANGES_REQUESTED** [review]

File: serde-support/src/main/java/io/micronaut/serde/support/LegacyBeansFactory.java (hunk 2 above)
Comment (dstepanov) [reply serde-support/src/main/java/io/micronaut/serde/support/LegacyBeansFactory.java:83]: Yes

File: serde-support/src/test/groovy/io/micronaut/serde/support/serdes/BeanContextSerdeSpec.groovy (hunk 3 above)
Comment (dstepanov) [reply serde-support/src/test/groovy/io/micronaut/serde/support/serdes/BeanContextSerdeSpec.groovy]: Added

**Review by graemerocher — APPROVED** [review]

**Pull request merged**